test/jdk/javax/swing/JButton/SpaceKeyActivatesButton.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.Robot;
+import java.awt.event.KeyEvent;
+import java.util.Arrays;
+import java.util.List;
+import javax.swing.JButton;
+import javax.swing.JFrame;
+import javax.swing.JPanel;
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
+import javax.swing.UnsupportedLookAndFeelException;
+
+import static java.util.stream.Collectors.toList;
+
+/*
+ * @test
+ * @key headful
+ * @summary Check whether pressing <Space> key generates
+ * ActionEvent on focused Button or not.
+ * @run main SpaceKeyActivatesButton
+ */
+public class SpaceKeyActivatesButton {
+    private static volatile boolean buttonPressed;
+    private static JFrame frame;
+
+    public static void main(String[] s) throws Exception {
+        runTest();
+    }
+
+    public static void runTest() throws Exception {
+        Robot robot = new Robot();
+        robot.setAutoDelay(100);
+        robot.setAutoWaitForIdle(true);
+        List<String> lafs = Arrays.stream(UIManager.getInstalledLookAndFeels())
+                                  .map(laf -> laf.getClassName())
+                                  .collect(toList());
+        for (String laf : lafs) {
+            try {
+                buttonPressed = false;
+                System.out.println("Testing L&F: " + laf);
+                SwingUtilities.invokeAndWait(() -> {
+                    setLookAndFeel(laf);
+                    createUI();
+                });
+
```

Comment:
I think you should add robot.waitForIdle() after the frame creation, and also wait until the button will get a focus, since the "requestFocusInWindow" might be ignored.

Reply:
Fixed it.